pkg/typesystem/reachabilitygraph.go

```diff
@@ -142,6 +144,78 @@ func (rg *ReachabilityGraph) RelationsEncounteredForResource(
 	return relationRefs, nil
 }
 
+// RelationsEncounteredForSubject returns all relations that are encountered when walking outward from a subject+relation.
+func (rg *ReachabilityGraph) RelationsEncounteredForSubject(
+	ctx context.Context,
+	allDefinitions []*core.NamespaceDefinition,
```

Comment:
nit: does this need to be "all definitions" or it could be any subset? The argument name would invite to think it has to be provided with all namespaces in the instance

Reply:
Its everything